azure-keyvault-secrets-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/boot/KeyVaultOperation.java

```diff
@@ -0,0 +1,69 @@
+package com.microsoft.azure.keyvault.spring.boot;
+
+import com.microsoft.azure.PagedList;
+import com.microsoft.azure.keyvault.KeyVaultClient;
+import com.microsoft.azure.keyvault.models.SecretBundle;
+import com.microsoft.azure.keyvault.models.SecretItem;
+
+import java.util.Collections;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class KeyVaultOperation {
+    private KeyVaultClient keyVaultClient;
+    private String vaultUri;
+
+    private ConcurrentHashMap<String, Object> propertyNamesHashMap;
+    private AtomicLong lastUpdateTime = new AtomicLong();
+    private final Object refreshLock = new Object();
+    private static final long CACHE_REFRESH_INTERVAL_IN_MS = 1800000L; // 30 minutes
+
+    public KeyVaultOperation(KeyVaultClient keyVaultClient, String vaultUri) {
+        this.keyVaultClient = keyVaultClient;
+        this.vaultUri = vaultUri;
+        if (vaultUri.endsWith("/")) {
+            this.vaultUri = vaultUri.substring(0, vaultUri.length() - 1);
+        }
+        createOrUpdateHashMap();
+        lastUpdateTime.set(System.currentTimeMillis());
+    }
+
+    public String[] list() {
+        return Collections.list(propertyNamesHashMap.keys()).toArray(new String[propertyNamesHashMap.size()]);
+    }
+
+    public Object get(String secretName) {
+        // NOTE: azure keyvault secret name convention: ^[0-9a-zA-Z-]+$ "." is not allowed
+        secretName = secretName.replace(".", "-");
+
+        // refresh periodically
+        if (System.currentTimeMillis() - lastUpdateTime.get() > CACHE_REFRESH_INTERVAL_IN_MS) {
+            synchronized (refreshLock) {
+                if (System.currentTimeMillis() - lastUpdateTime.get() > CACHE_REFRESH_INTERVAL_IN_MS) {
+                    lastUpdateTime.set(System.currentTimeMillis());
+                    // refresh propertyNames
+                    createOrUpdateHashMap();
+                }
+            }
+        }
+
+        if (propertyNamesHashMap.containsKey(secretName)) {
+            final SecretBundle secretBundle = keyVaultClient.getSecret(vaultUri, secretName);
+            return secretBundle.value();
```

Comment:
to cache them together, when update cache, need to call keyvalut multiple times to retrieve all property value one by one. if property list is long, time is matter, and one get call to keyvault is < 1s. so not caching values together. list from to keyvault cost about 3 seconds, make caching propertyname list more useful.